django/chat/utils.py

```diff
@@ -212,12 +212,9 @@ def sse_string(message: str, format=True, dots=False, remove_stop=False) -> str:
 
         yield sse_string(full_message, format, dots=False, remove_stop=True)
 
-        # Response is done! Update costs and save message
+        await sync_to_async(llm.create_costs)()
+
         message = await sync_to_async(Message.objects.get)(id=message_id)
-        user = await sync_to_async(lambda: chat.user)()
-        costs = await sync_to_async(llm.create_costs)(user, message.mode)
-        total_cost = sum([cost.usd_cost for cost in costs])
-        message.cost = total_cost
```

Comment:
The message and document cost properties are now calculated / set from within the Cost.objects.new() function.